### PR TITLE
Adds identities on marshaling function for G1,G2,GT.

### PR DESCRIPTION
**Pull request by armfazh**
Includes tests that verify the issue mentioned in #8.

### CODE REVIEW

File: bn256.go [hunk 1]
```diff
@@ -105,6 +105,11 @@ func (e *G1) Marshal() []byte {
 	// Each value is a 256-bit number.
 	const numBytes = 256 / 8
 
+	if e.p == nil {
+		e.p = &curvePoint{}
+		e.p.SetInfinity()
+	}
+
```
Comment (Bren2010) [inline bn256.go:111]: This isn't necessary on G1 or G2 because of the MakeAffine below

File: bn256_test.go [hunk 2]
```diff
@@ -41,6 +41,18 @@ func TestG1Marshal(t *testing.T) {
...
+	ma = Ga.Marshal()
+
+	one := &curvePoint{}
+	one.SetInfinity()
+	Gb = &G1{p: one}
+	mb = Gb.Marshal()
+
+	if !bytes.Equal(ma, mb) {
+		t.Fatal("bytes are different")
+	}
```
Comment (Bren2010) [inline bn256_test.go]: I don't want these tests

**Review by Bren2010 — CHANGES_REQUESTED** [review]

File: bn256.go (hunk 1 above)
Comment (armfazh) [reply bn256.go:111]: Right, now is fixed.

File: bn256_test.go (hunk 2 above)
Comment (armfazh) [reply bn256_test.go]: removed.

**Review by Bren2010 — APPROVED** [review]

**Pull request merged**